server/resources/resource_management_test.go

```diff
@@ -32,40 +32,64 @@ import (
 	"github.com/cloudwan/gohan/util"
 	"github.com/mohae/deepcopy"
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
 	"github.com/twinj/uuid"
 )
 
 var _ = Describe("Resource manager", func() {
 	const (
-		resourceID1 = "6660fbf8-ca60-4cb0-a42e-9a913beafbaf"
-		resourceID2 = "6660fbf8-ca60-4cb0-a42e-9a913beafbae"
+		adminResourceID       = "6660fbf8-ca60-4cb0-a42e-9a913beafbaf"
+		memberResourceID      = "6660fbf8-ca60-4cb0-a42e-9a913beafbae"
+		otherDomainResourceID = "6660fbf8-ca60-4cb0-a42e-9a913beafbad"
 	)
 
 	var (
-		manager       *schema.Manager
-		adminAuth     schema.Authorization
-		memberAuth    schema.Authorization
-		auth          schema.Authorization
-		context       middleware.Context
-		schemaID      string
-		path          string
-		action        string
-		currentSchema *schema.Schema
-		extensions    []*schema.Extension
-		env           extension.Environment
-		events        map[string]string
-		timeLimit     time.Duration
-		timeLimits    []*schema.PathEventTimeLimit
-		ctx           context_pkg.Context
+		manager                 *schema.Manager
+		adminAuth               schema.Authorization
+		memberAuth              schema.Authorization
+		domainScopedAuth        schema.Authorization
+		auth                    schema.Authorization
+		context                 middleware.Context
+		schemaID                string
+		path                    string
+		action                  string
+		currentSchema           *schema.Schema
+		extensions              []*schema.Extension
+		env                     extension.Environment
+		events                  map[string]string
+		timeLimit               time.Duration
+		timeLimits              []*schema.PathEventTimeLimit
+		ctx                     context_pkg.Context
+		adminResourceData       map[string]interface{}
+		memberResourceData      map[string]interface{}
+		otherDomainResourceData map[string]interface{}
 	)
 
 	BeforeEach(func() {
 		manager = schema.GetManager()
 		ctx = context_pkg.Background()
 
-		adminAuth = schema.NewAuthorization(adminTenantID, "admin", adminTokenID, []string{"admin"}, nil)
-		memberAuth = schema.NewAuthorization(memberTenantID, "demo", memberTokenID, []string{"Member"}, nil)
+		domainA := schema.Domain{
+			ID:   domainAID,
+			Name: "domainA",
+		}
+
+		adminAuth = schema.NewAuthorizationBuilder().
```

Comment:
If you mean `admin` here as cloud administrator or super-user to see resources across domains, we should use `admin_project` concept here for v3 auth. https://developer.openstack.org/api-ref/identity/v3/index.html#authentication-and-token-management